packages/flutter/lib/src/widgets/pages.dart

```diff
@@ -60,18 +69,19 @@ class PageRouteBuilder<T> extends PageRoute<T> {
     this.barrierLabel,
     this.maintainState = true,
     bool fullscreenDialog = false,
-  }) : assert(pageBuilder != null),
+  }) : assert((bodyBuilder ?? pageBuilder) != null),
```

Comment:
this wouldn't catch both of them being non-null

you want something like
```
  assert(bodyBuilder == null || pageBuilder == null),
  assert(bodyBuilder != null || pageBuilder != null),
```
...or:
```
  assert((bodyBuilder == null) != (pageBuilder == null)),
```

Reply:
Do we want to block both of them being non-null? it will prefer bodyBuilder in this case.